doc/source/data-ingest.rst

```diff
@@ -0,0 +1,140 @@
+   An entities location describes where in space the entity exists.  Locations are defined in units according to the coordinate system in which they are displayed. (Latitude and Longitude on a world map.)
+**Time**
+   Time defines the moment in time at which the rest of the information describing an entity is valid.
+
+With this information Conduce can build a representation of an object that exists in space and time.  Entity definitions can be more complex, for more information on how to build entities please read :doc:`conduce-entities`.
+
+--------
+Datasets
+--------
+
+Datasets are the containers that hold entities.  A dataset may be thought of as a table.  Each entity in a dataset is similar in nature, but has unique properties.  In order to represent and entity in Conduce, its sample data must be ingested into a dataset.  In order to perform any API operations it is recommended that you generate an API key.  (See :doc:`api-key-creation` for more information.)  The first step in ingesting data is acquiring a dataset ID.  If you are ingesting data into a new dataset you'll need to create the dataset  before entities can be ingested.
+
+:py:func:`api.create_dataset`::
+
+    world_cities_dataset_id = conduce.api.create_dataset('world cities', host='app.conduce.com', api-key='00000000-0000-0000-0000-000000000000')
+
+The dataset ID returned is a UUID.  It will be passed as a function parameter to the :py:func:`api.ingest_samples` function.
+
+If you want to ingest data into an existing dataset you will need to acquire its dataset ID.  Datasets are Conduce resources.  All resources have an ID and can be queried with the resource API.  The Conduce Python API provides convenience methods for listing datasets
+
+:py:func:`api.list_datasets`::
+
+    datasets = conduce.api.list_datasets(host='app.conduce.com', api-key='00000000-0000-0000-0000-000000000000')
+
+This call returns a list of dictionaries, each contains information about a particular dataset::
+
+    {
+       "name": "US Cities",
+       "tags": [],
+       "create_time": 1506544998452,
+       "version": 0,
+       "mime": "application/json",
+       "modify_time": 1508869186664,
+       "revision": 4,
+       "type": "DATASET",
+       "id": "a262342b-22d2-4368-4698-cf75b62b7cb3",
+       "size": 2
+     }
+
+This dataset object contains information to help identify a dataset.  The fields ``name`` and ``tags``, along with ``create_time`` and ``modify_time`` are particularly useful for identifying datasets.  The :py:func:`api.find_datasets` function can be used to list only datasets that match certain search criteria.
+
+-------------------
+Populating datasets
+-------------------
+
+Once a dataset has been created, it needs to be populated before it can be visualized.  Datasets are populated by making API calls to create entities.  An entity is a person, place, or thing that exists in space and time.  See :doc:`conduce-entities` for detailed information on creating entities.
```

Comment:
I think this is confusing.  We don't create entities, we add samples.  This makes me think as a user I'll have to run a create entities api call or something.

Reply:
Maybe this should say "Static datasets are populated by making API calls to add entities.  For dynamic datasets, there is no need to create entities - instead users make API calls to add samples"

Reply:
What if I change create to add.  I agree we are not really creating.  Entities are created by the user and added to the dataset.

Reply:
I think there is a terminology problem.  Entities are the concept.  They are not directly represented in the dataset.  Yet I am using language that implies that is the case.  In reality only samples exist in the dataset.  Samples define the entity.  However we've restricted use of the word sample to only those objects that have a time component.  That leaves us with the question of what to call the object that defines a timeless/infinite entity.

Any ideas?

Reply:
I can't think of any great words, though maybe something like static/timeless/infinite ___ works - static entity, static sample, infinite sample?

Alternatively, we could rename entity.  I recall something along the lines of "we represent unique objects using entities" somewhere, we could just call them objects.  Objects may be ok for the timeless things too, but I don't think it captures the timeless part well.

Reply:
The intro says "a constant continuous function (constant entity)" - something related to that maybe?